### DISCUSSION TITLE
How will JS code look like in unpacked APK built with Static Hermes?

### DISCUSSION THREAD

**Discussion by IslamRustam** [facebook/hermes#1167] [Static Hermes]
Hello!

As you might remember, I started this discussion a while ago https://github.com/facebook/hermes/discussions/1152. There I found out that if we create APK with JSC - we get index.android.bundle, which is basically minified JS. If we bundle with Hermes - we will get the same file, but it's going to be in bytecode.

If I, let's say, create an APK with Static Hermes, what am I theoretically going to get? Judging from Mikov's presentation from React Native EU 2023 I will get index.android.bundle with assembler-like code. Or something else?

**Comment by tmikov** [comment]
You will get a bytecode bundle and probably a natively compiled .so for the parts that you have chosen to compile to native. We still haven't quite ironed out the details of how it will be packaged.

**Reply by IslamRustam** [reply]
I see. In Flutter you also get a .so files which cannot be viewed normally in IDE. 

Could you please explain, what you mean by natively compiled? It will turn selected parts of JS into C/Assembler/Bytecode of C? 

I'm not really proficient in theory of compilators, so might not understand much.

**Reply by tmikov** [reply]
It will contain binary instructions executable by the CPU of that device.
